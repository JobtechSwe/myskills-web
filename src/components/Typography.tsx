import React from 'react'
import styled from '@emotion/styled'
import {
  color,
  ColorProps,
  fontFamily,
  FontFamilyProps,
  fontSize,
  FontSizeProps,
  fontWeight,
  FontWeightProps,
  textAlign,
  TextAlignProps,
  space,
  SpaceProps,
} from 'styled-system'

type TypographyProps = ColorProps &
  FontFamilyProps &
  FontSizeProps &
  FontWeightProps &
  ColorProps &
  TextAlignProps &
  SpaceProps

type HeadingProps = React.HTMLProps<HTMLHeadingElement> & TypographyProps
type ParagraphProps = React.HTMLProps<HTMLParagraphElement> & TypographyProps

export const H1 = styled.h1<HeadingProps>`
  ${color}
  ${fontFamily}
  ${fontSize}
  ${fontWeight}
  ${textAlign}
  ${space}
`

export const H2 = styled.h2<HeadingProps>`
  ${color}
  ${fontFamily}
  ${fontSize}
  ${fontWeight}
  ${textAlign}
  ${space}
`

export const H3 = styled.h3<HeadingProps>`
  ${color}
  ${fontFamily}
  ${fontSize}
  ${fontWeight}
  ${textAlign}
  ${space}
`

export const Paragraph = styled.p<ParagraphProps>`
  ${color}
  ${fontFamily}
  ${fontSize}
  ${fontWeight}
  ${textAlign}
  ${space}
`

const common: HeadingProps = {
  color: 'persianBlue',
  fontFamily: 'default',
}

H1.defaultProps = {
  ...common,
  fontSize: ['small', 'medium', 'large'],
  fontWeight: 600,
}

H2.defaultProps = {
  ...common,
  fontSize: ['small', 'medium'],
}

H3.defaultProps = {
  ...common,
  fontSize: 'small',
}

Paragraph.defaultProps = {
  fontFamily: 'default',
  fontSize: 'small',
  mt: 'small',
  mb: 'medium',
}
