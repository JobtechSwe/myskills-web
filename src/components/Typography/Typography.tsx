import React from 'react'
import styled from '@emotion/styled'
import { theme } from '../../theme'
import {
  color,
  ColorProps,
  fontSize,
  FontSizeProps,
  space,
  SpaceProps,
} from 'styled-system'

export const H1 = styled.h1`
  ${color}
  ${fontSize}
  ${space}
`
export const H2 = styled.h2`
  ${color}
  ${fontSize}
  ${space}
`
export const H3 = styled.h3`
  ${color}
  ${fontSize}
  ${space}
`

export const Paragraph = styled.p`
  ${color}
  ${fontSize}
  ${space}
`

type HeadingProps = ColorProps & FontSizeProps & SpaceProps

const common: HeadingProps = {
  color: theme.colors.white,
  mt: theme.space.none,
}

H1.defaultProps = {
  ...common,
  fontSize: ['small', 'medium', 'large'],
}

H2.defaultProps = {
  ...common,
  fontSize: ['small', 'medium'],
}

H3.defaultProps = {
  ...common,
  fontSize: 'small',
}

Paragraph.defaultProps = {
  fontSize: 'small',
}
