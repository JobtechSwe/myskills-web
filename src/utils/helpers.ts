import Cookies from 'js-cookie'
import { LocalStateProps, ContactInformationProps } from '../graphql/client'
import {
  EducationInput,
  Occupation,
  OntologyRelationResponse,
} from '../generated/myskills'

export const getCookie = (name: string) => Cookies.get(name)

export const setCookie = (
  name: string,
  value: string,
  options: object = {}
) => {
  Cookies.set(name, value, options)
}

export const removeCookie = (name: string) => {
  Cookies.remove(name)
}

export const redirect = (route: string) => (location.href = route)

interface StorageEntryProps {
  type: string
}

interface Educations extends StorageEntryProps {
  type: 'educations'
  data: EducationInput[]
}

interface OccupationStorage extends StorageEntryProps {
  type: 'occupation'
  data: Occupation
}

interface Skills extends StorageEntryProps {
  type: 'skills'
  data: OntologyRelationResponse[]
}
interface Traits extends StorageEntryProps {
  type: 'traits'
  data: string[]
}
interface WhoAmI extends StorageEntryProps {
  type: 'whoAmI'
  data: string
}

interface ContactInformation extends StorageEntryProps {
  type: 'contact'
  data: ContactInformationProps
}

export type StorageEntry =
  | ContactInformation
  | Educations
  | OccupationStorage
  | Skills
  | Traits
  | WhoAmI

export const storageHelper = {
  load: (initialState: LocalStateProps) =>
    Object.entries(initialState).reduce(
      (acc, [prop, val]) => ({
        ...acc,
        [prop]: JSON.parse(localStorage.getItem(prop) as string) || val,
      }),
      {} as LocalStateProps
    ),
  set: (payload: StorageEntry) =>
    localStorage.setItem(payload.type, JSON.stringify(payload.data)),
}
