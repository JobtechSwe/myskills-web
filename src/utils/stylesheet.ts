const colors = {
  lassekongo: 'dodgerblue',
  primary: 'pink',
}

const sizes = ['0', '1rem', '2rem', '4rem']

const typography = {
  default: 'Arial',
}

const space = {
  none: sizes[0],
  small: sizes[1],
  medium: sizes[2],
  big: sizes[3],
}

export default { colors, space, typography }
