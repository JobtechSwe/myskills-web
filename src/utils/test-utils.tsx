import * as React from 'react'
import ApolloClient from 'apollo-client'
import { ApolloProvider } from 'react-apollo-hooks'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { MockLink, MockedProvider } from 'react-apollo/test-utils'
import { ThemeProvider } from 'emotion-theming'
import { render as rtlRender } from 'react-testing-library'
import { theme } from 'theme'

export const render = (children: React.ReactNode, mocks?: any[]) => {
  const client = new ApolloClient({
    cache: new InMemoryCache({ addTypename: true }),
    link: new MockLink(mocks || []),
  })

  return rtlRender(
    <MockedProvider addTypename={true} mocks={mocks}>
      <ApolloProvider client={client}>
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      </ApolloProvider>
    </MockedProvider>
  )
}
