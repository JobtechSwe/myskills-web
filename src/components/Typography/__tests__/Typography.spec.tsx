import * as React from 'react'
import { H1, H2, H3, Paragraph } from '../Typography'
import { render } from '../../../utils/test-utils'

describe('components/Typography', () => {
  it('renders an H1', async () => {
    const { container } = render(<H1>Heading One</H1>)

    expect(container).toMatchSnapshot()
  })

  it('renders an H2', async () => {
    const { container } = render(<H2>Heading Two</H2>)

    expect(container).toMatchSnapshot()
  })

  it('renders an H3', async () => {
    const { container } = render(<H3>Heading Three</H3>)

    expect(container).toMatchSnapshot()
  })

  it('renders a Heading with props', async () => {
    const { container } = render(
      <H3 color="red" fontSize="small" mt="15">
        Heading Three
      </H3>
    )

    expect(container).toMatchSnapshot()
  })

  it('renders a Paragraph', async () => {
    const { container } = render(<Paragraph>A paragraph.</Paragraph>)

    expect(container).toMatchSnapshot()
  })

  it('renders a Paragraph with props', async () => {
    const { container } = render(
      <Paragraph color="red" fontSize="small" m="15">
        Heading Three
      </Paragraph>
    )

    expect(container).toMatchSnapshot()
  })
})
