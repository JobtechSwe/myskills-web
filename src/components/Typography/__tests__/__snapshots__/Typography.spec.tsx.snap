// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Typography renders a Heading with props 1`] = `
<div>
  <h3
    class="css-q4e2n5"
    color="red"
    font-size="small"
  >
    Heading Three
  </h3>
</div>
`;

exports[`components/Typography renders a Paragraph 1`] = `
<div>
  <p
    class="css-3v9ikz"
    font-size="small"
  >
    A paragraph.
  </p>
</div>
`;

exports[`components/Typography renders a Paragraph with props 1`] = `
<div>
  <p
    class="css-1jpqnab"
    color="red"
    font-size="small"
  >
    Heading Three
  </p>
</div>
`;

exports[`components/Typography renders an H1 1`] = `
<div>
  <h1
    class="css-1h5f8tm"
    color="white"
    font-size="small,medium,large"
  >
    Heading One
  </h1>
</div>
`;

exports[`components/Typography renders an H2 1`] = `
<div>
  <h2
    class="css-1qu5h0y"
    color="white"
    font-size="small,medium"
  >
    Heading Two
  </h2>
</div>
`;

exports[`components/Typography renders an H3 1`] = `
<div>
  <h3
    class="css-1bs1ybp"
    color="white"
    font-size="small"
  >
    Heading Three
  </h3>
</div>
`;
